c++/src/kj/refcount.h

```diff
@@ -115,6 +132,93 @@ Own<T> Refcounted::addRefInternal(T* object) {
   return Own<T>(object, *refcounted);
 }
 
+template <typename T>
+Rc<T> Refcounted::addRcRefInternal(T* object) {
+  Refcounted* refcounted = object;
+  ++refcounted->refcount;
+  return Rc<T>(object);
```

Comment:
Are there any challenges here if `this` wasn't actually created by Rc<T>::create? For instance, if T is allocated on the stack? Might be a good idea to add a check in here that the thing is actually refcounted?